test/cri-containerd/disable_vpmem_test.go

```diff
@@ -0,0 +1,110 @@
+//go:build functional
+// +build functional
+
+package cri_containerd
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/Microsoft/hcsshim/pkg/annotations"
+	runtime "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+)
+
+// Use unique names for pods & containers so that if we run this test multiple times in parallel we don't
+// get failures due to same pod/container names.
+func uniqueRef() string {
+	t := time.Now()
+	var b [3]byte
+	// Ignore read failures, just decreases uniqueness
+	rand.Read(b[:])
+	return fmt.Sprintf("%d-%s", t.UnixNano(), base64.URLEncoding.EncodeToString(b[:]))
+}
+
+func Test_70LayerImagesWithNoVPmemForLayers(t *testing.T) {
+	requireFeatures(t, featureLCOW)
+
+	ubuntu70Image := "cplatpublic.azurecr.io/ubuntu70extra:18.04"
+	alpine70Image := "cplatpublic.azurecr.io/alpine70extra:latest"
+	testImages := []string{ubuntu70Image, alpine70Image}
+	pullRequiredLCOWImages(t, []string{imageLcowK8sPause, ubuntu70Image, alpine70Image})
+
+	client := newTestRuntimeClient(t)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	nContainers := 4
+	podID := ""
+	containerIDs := make([]string, nContainers)
+
+	defer cleanupPod(t, client, ctx, &podID)
+	for i := 0; i < nContainers; i++ {
+		defer cleanupContainer(t, client, ctx, &containerIDs[i])
+	}
+
+	sandboxRequest := getRunPodSandboxRequest(
+		t,
+		lcowRuntimeHandler,
+		WithSandboxAnnotations(map[string]string{
+			// 1 VPMEM device will be used for UVM VHD and all container layers
+			// will be attached over SCSI.
+			annotations.VPMemCount: "1",
```

Comment:
Is this for the initrd VHD? AFAIK PMem is not used for that. Can we set this to 0?

Reply:
I thought if the rootfs.vhd can be found we plop it on the first pmem slot? https://github.com/microsoft/hcsshim/blob/master/internal/uvm/create_lcow.go#L562

https://github.com/microsoft/hcsshim/blob/master/internal/uvm/create_lcow.go#L159-L163

Reply:
I thought if the rootfs.vhd can be found we plop it on the first pmem slot? https://github.com/microsoft/hcsshim/blob/master/internal/uvm/create_lcow.go#L562

https://github.com/microsoft/hcsshim/blob/master/internal/uvm/create_lcow.go#L159-L163

Reply:
Right, I forgot that we only pass the initrd directly via KernelDirectBoot for the non-VHD case. Thanks Danny.

Reply:
Changed this to 0.